### PR TITLE
Feature angular16

### PR DESCRIPTION
**Pull request by mhenkens**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/ngx-form-errors/blob/master/CONTRIBUTING.md#-commit-message-guidelines
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: demo-app/ng16/tsconfig.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "@nationalbankbelgium/code-style/tsconfig/4.4.x/ng13",
+  "extends": "@nationalbankbelgium/code-style/tsconfig/4.9.x/ng15",
```
Comment (SuperITMan) [inline demo-app/ng16/tsconfig.json:2]:
```suggestion
  "extends": "@nationalbankbelgium/code-style/tsconfig/5.1.x/ng16",
```

File: demo-app/ng16/tsconfig.json [hunk 2]
```diff
@@ -11,9 +11,10 @@
     "module": "es2020",
     "moduleResolution": "node",
     "importHelpers": true,
-    "target": "es2020",
+    "target": "ES2022",
```
Comment (SuperITMan) [inline demo-app/ng16/tsconfig.json:14]: Is it required to override the `target` as well as the `lib` ?

File: package.json [hunk 3]
```diff
@@ -77,7 +77,7 @@
     "stylelint-config-prettier": "^8.0.2",
     "tslib": "~2.3.0",
     "typescript": "~4.9.5",
-    "zone.js": "~0.11.4"
+    "zone.js": "~0.13.3"
   },
   "peerDependencies": {
     "@angular/common": "13.x || 14.x",
```
Comment (SuperITMan) [inline package.json:83]: I think the peerDeps should be upgraded

```suggestion
    "@angular/common": "15.x || 16.x",
    "@angular/core": "15.x || 16.x",
    "@angular/forms": "15.x || 16.x",
    "rxjs": "7.x"
```

**Review by SuperITMan — CHANGES_REQUESTED** [review]

File: demo-app/ng15/tsconfig.json [hunk 4]
```diff
@@ -11,15 +11,9 @@
...
-      "node_modules/@types"
-    ],
-    "lib": [
-      "dom",
-      "dom.iterable",
-      "es2017"
-    ],
+    "target": "es2022",
+    "typeRoots": ["node_modules/@types"],
+    "lib": ["dom", "dom.iterable", "es2017"],
```
Comment (SuperITMan) [inline demo-app/ng15/tsconfig.json:16]: Should not it be es2022 instead of es2017 ?

File: demo-app/ng16/tsconfig.json [hunk 5]
```diff
@@ -11,9 +11,10 @@
...
     "typeRoots": ["node_modules/@types"],
-    "lib": ["dom", "dom.iterable", "es2017"]
+    "lib": ["dom", "dom.iterable", "es2017"],
```
Comment (SuperITMan) [inline demo-app/ng16/tsconfig.json:16]: Should not it be es2022 instead of es2017 ?

**Review by SuperITMan — CHANGES_REQUESTED** [review]

**Review by SuperITMan — CHANGES_REQUESTED** [review]

File: package.json [hunk 6]
```diff
@@ -36,20 +36,20 @@
...
-    "@angular/platform-browser": "^15.2.10",
-    "@angular/platform-browser-dynamic": "^15.2.10",
-    "@angular-devkit/build-angular": "^15.2.11",
+    "@angular/cli": "^16.2.15",
+    "@angular/common": "^16.2.12",
+    "@angular/compiler": "^16.2.12",
+    "@angular/compiler-cli": "^16.2.12",
+    "@angular/core": "^16.2.12",
+    "@angular/platform-browser": "^16.2.12",
+    "@angular/forms": "^16.2.12",
```
[the remark this answers is not in the document]
Comment (mhenkens) [reply package.json:45]: May be the order I select the line when merge conflict

**Review by SuperITMan — APPROVED** [review]

**Pull request merged**